### PR TITLE
use unique_ptr for map_update_thread

### PR DESCRIPTION
**Pull request by gezp**
replace raw pointer with unique_ptr.

### CODE REVIEW

File: nav2_costmap_2d/src/costmap_2d_ros.cpp [hunk 1]
```diff
@@ -238,10 +238,10 @@ Costmap2DROS::on_activate(const rclcpp_lifecycle::State & /*state*/)
   stopped_ = true;  // to active plugins
   stop_updates_ = false;
   map_update_thread_shutdown_ = false;
-
-  map_update_thread_ = new std::thread(
-    std::bind(
-      &Costmap2DROS::mapUpdateLoop, this, map_update_frequency_));
+  map_update_thread_ = std::make_unique<std::thread>(
+    [this]() {
```
Comment (SteveMacenski) [inline nav2_costmap_2d/src/costmap_2d_ros.cpp:242]: Why not bind?

Comment (SteveMacenski) [reply nav2_costmap_2d/src/costmap_2d_ros.cpp:242]: I'd prefer that, if it works

Comment (gezp) [reply nav2_costmap_2d/src/costmap_2d_ros.cpp:242]: it seems that lambda is better than bind, you could see more detail here https://github.com/ros-planning/moveit2/issues/872. but if you still think we should use bind, i can change back to bind.

Comment (SteveMacenski) [reply nav2_costmap_2d/src/costmap_2d_ros.cpp:242]: Given its a 1 time setup, I don't think the overhead is a big deal

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**